openlibrary/plugins/upstream/borrow.py

```diff
@@ -141,16 +117,26 @@ def POST(self, key):
         if not edition:
             raise web.notfound()
 
-        error_redirect = edition.url("/borrow")
+        # Make a call to availability v2
+        # update the subjects according to result
+        # if `open`, redirect to bookreader
+        olid = key.split('/')[-1]
+        response = lending.get_edition_availability(olid)
+        availability = response[olid] if response else {}
+        ocaid = availability.get('identifier')
+        if availability and availability['status'] == 'open':
+            raise web.seeother('https://archive.org/stream/' + ocaid)
```

Comment:
I don't believe so -- in the existing `make_bookreader_auth_link` method, ocaid is used as is.